### ISSUE TITLE
Multiple group block separating into individual groups

### ISSUE THREAD

**Issue by shoutsid** [joonty/pessimize#11]
Hope this helps some what! Just a though and what I expected from the result.
*When I have:*
```ruby
group :production, :staging, :live_dev do
  gem 'newrelic_rpm'
  gem 'rails_12factor'
  gem 'heroku-deflater'
end
```

*What I get after running `pessimize`*
```ruby
group :production do
  gem 'newrelic_rpm', '~> 3.12'
  gem 'rails_12factor', '~> 0.0'
  gem 'heroku-deflater', '~> 0.5'
end

group :staging do
  gem 'newrelic_rpm', '~> 3.12'
  gem 'rails_12factor', '~> 0.0'
  gem 'heroku-deflater', '~> 0.5'
end

group :live_dev do
  gem 'newrelic_rpm', '~> 3.12'
  gem 'rails_12factor', '~> 0.0'
  gem 'heroku-deflater', '~> 0.5'
end
```

This will cause the following errors to appear on any `bundle` command and will repeat for each group stage.
```
Your Gemfile lists the gem newrelic_rpm (~> 3.12) more than once.
You should probably keep only one of them.
While it's not a problem now, it could cause errors if you change the version of just one of them later.
Your Gemfile lists the gem rails_12factor (~> 0.0) more than once.
You should probably keep only one of them.
While it's not a problem now, it could cause errors if you change the version of just one of them later.
Your Gemfile lists the gem heroku-deflater (~> 0.5) more than once.
You should probably keep only one of them.
```

*What I expect*
```ruby
group :production, :staging, :live_dev do
  gem 'newrelic_rpm', '~> 3.12'
  gem 'rails_12factor', '~> 0.0'
  gem 'heroku-deflater', '~> 0.5'
end
```

**Comment by joonty** [comment]
Hi @shoutsid! Great to hear from you.

You should check out the `token-parsing` branch, which should fix the problem. Basically, the master branch uses the old parsing method, which loads the Gemfile and hijacks the method calls to try and work out how to piece it all back together again, but some information gets lost (e.g. comments, the position of original code fragments, etc). The `token-parsing` branch splits the Gemfile up into tokens and can keep the exact position of the original tokens.

It's not perfect, hence why I haven't merged it into master. It works for 99% of cases, but if you had some unusual but perfectly valid syntax in your Gemfile then it would throw a wobbler. I've got a plan to implement a better parser soon, just need a bit of time ;)

I'll keep this issue open until I can get something working and release a new gem, because it's technically valid for the most recent version.

Thanks

**Comment by joonty** [comment]
Hi again @shoutsid,

This has been fixed in version 0.2.0 and even more fixed (hehe) in the very latest version 0.3.0.

This version should keep the Gemfile syntax exactly the same, but insert or replace versions as expected.

Thanks
